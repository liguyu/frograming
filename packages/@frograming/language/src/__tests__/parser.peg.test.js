import { readFile, readdirSync } from 'fs-extra';
import { join } from 'path';
import { parse } from '../parser.peg.js';

const readdirFullSync = (dir) => readdirSync(dir).map(name => join(dir, name));

describe('pegParser', () => {
  readdirFullSync(join(__dirname, './frogram-specs/frogs'))
    .filter(n => /.frog$/.test(n))
    .forEach(path => it(path, async () => {
      expect(parse(await readFile(path, 'utf8'))).toMatchSnapshot();
    }));
});
