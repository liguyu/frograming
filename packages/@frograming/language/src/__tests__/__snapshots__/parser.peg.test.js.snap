// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pegParser /Users/ycmjason/Development/frograming/packages/@frograming/language/src/__tests__/frogram-specs/frogs/01.frog 1`] = `
Array [
  "moveDown",
  "moveUp",
  "moveDown",
  "moveDown",
  Object {
    "!isCarUp&&isCarLeft||isLogUp": Array [
      "moveDown",
      "moveLeft",
      "moveUp",
      Object {
        "else": Array [
          "moveLeft",
        ],
        "isCarUp": Array [
          "moveUp",
        ],
      },
      "moveLeft",
      "moveUp",
      Object {
        "else": Array [
          "moveLeft",
        ],
        "isCarUp": Array [
          "moveUp",
        ],
      },
    ],
    "else": Array [
      "moveUp",
    ],
  },
]
`;
